### PR TITLE
feat: new Dashboard Navigation

### PR DESCRIPTION
**Pull request by piotr-pajak**
## Jira issue(s)
[LC-423](https://selleolabs.atlassian.net/browse/LC-423)

## Preview
Screenshot 2024-11-29 at 15 22 31
Screenshot 2024-11-29 at 15 22 25
Screenshot 2024-11-29 at 15 22 20
Screenshot 2024-11-29 at 15 22 12
Screenshot 2024-11-29 at 15 22 02

[LC-423]: https://selleolabs.atlassian.net/browse/LC-423?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/web/app/components/Navigation/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  const { isAdmin, isTutor, role } = useUserRole();
+
+  return (
+    <header className="sticky top-0 w-full z-10 h-min 2xl:static 2xl:py-4 2xl:px-2 2xl:w-14 3xl:w-60 2xl:h-dvh 2xl:flex 2xl:flex-col 2xl:gap-y-6 3xl:p-4">
+      <NavigationHeader isMobileNavOpen={isMobileNavOpen} setIsMobileNavOpen={setIsMobileNavOpen} />
+      <Separator className="sr-only 2xl:not-sr-only 2xl:h-[1px] bg-primary-200" />
+      <nav
+        className={cn(
+          "2xl:flex 2xl:flex-col 2xl:justify-between 2xl:h-full",
+          isMobileNavOpen
```
Comment (typeWolffo) [inline apps/web/app/components/Navigation/Navigation.tsx:26]: I think the typical classmerger function object notation will be more readable than ternary

File: apps/web/app/config/navigationConfig.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export interface ParentMenuItem extends BaseMenuItem {
   children: MenuItemType[];
 }
 
-export type MenuItemType = LeafMenuItem | ParentMenuItem;
+export type MenuItemType = LeafMenuItem;
```
Comment (typeWolffo) [inline apps/web/app/config/navigationConfig.ts:20]: If we don't need recursive menu items anymore, can you delete the unused types above?

**Review by typeWolffo — COMMENTED** [review]
I left a few comments

**Review by wielopolski — APPROVED** [review]
LGTM 🍻

**Pull request merged**